build.gradle

```diff
@@ -106,15 +119,39 @@ dependencies {
     // These dependencies are only needed for running tests
 
     testImplementation 'org.hamcrest:hamcrest:2.2'
-    testImplementation "org.junit.jupiter:junit-jupiter-api:5.6.2"
-    testRuntimeOnly "org.junit.jupiter:junit-jupiter-engine:5.6.2"
+    testImplementation "org.junit.jupiter:junit-jupiter-api:5.6.+"
```

Comment:
I don't like the wild-card `+` in the version. I know it gives us patches for free, but on the other hand we don't have **reproducible builds** anymore (not sure we even have them....). 

I like the idea that building the launcher now, tomorrow, or in a year from this commit will yield the same results. It's not critical, though, and I leave this just as a comment for your information :nerd_face:

Reply:
In that case, you're interested in https://docs.gradle.org/current/userguide/dependency_locking.html

To a large degree, the variability is something we can reduce by not using version ranges in our direct dependencies, yes. But the lockfile is necessary to prevent variation in transitive dependencies.

"patches for free" would also be a less attractive option if I knew how one was supposed to get patches for _cheap_ in the gradle-verse, but as it is I have no idea how one is supposed to know which things are out of date and what they should be updated to.